src/LanguageServer/Microsoft.CodeAnalysis.FileBasedPrograms.Features/PropertyAppDirectiveCompletionProvider.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Completion;
+using Microsoft.CodeAnalysis.CSharp.Completion.Providers;
+
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.FileBasedPrograms;
+
+[ExportCompletionProvider(nameof(PropertyAppDirectiveCompletionProvider), LanguageNames.CSharp)]
+[Shared]
+[method: ImportingConstructor]
+//ImportingConstructorMessage
+// [method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+[method: Obsolete("TODO2 properly share constant message", error: true)]
+internal class PropertyAppDirectiveCompletionProvider() : LSPCompletionProvider
+{
+    private const string DirectiveKind = "property";
+
+    public override bool IsInsertionTrigger(SourceText text, int characterPosition, CompletionOptions options)
+    {
+        return TriggerCharacters.Contains(text[characterPosition])
+            || (options.TriggerOnTypingLetters && CompletionUtilities.IsStartingNewWord(text, characterPosition));
+    }
+
+    public override ImmutableHashSet<char> TriggerCharacters { get; } = [':'];
+
+    internal override string Language => LanguageNames.CSharp;
+
+    public override async Task ProvideCompletionsAsync(CompletionContext context)
+    {
+        var tree = await context.Document.GetSyntaxTreeAsync(context.CancellationToken).ConfigureAwait(false);
+        if (tree is null || !tree.Options.Features.ContainsKey("FileBasedProgram"))
```

Comment:
you exported for C# so GetSyntaxTreeAsync can't return null.  We have GetRequiredSyntaxTreeAsync for this sort of case.